### PR TITLE
Mg nav2 (finished nav stories)

### PR DESCRIPTION
**Pull request by Gallup93**
Finished user stories 2-9

created controllers for merch and errors

created functionality to guard against users accessing links they dont have permission to

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -14,4 +14,13 @@
   describe "enum" do
     it { should define_enum_for(:role).with([:default, :merchant, :admin]) }
   end
+
+  describe "roles" do
+    it "can be created as a default user" do
+      default_1 = User.create(name: "Hank Hill", address: "801 N Alamo St", city: "Arlen", state: "Texas",
+                             zip: "61109", email: "ProPAIN@aol.com", password_digest: "W33dWacker", role: 0)
```
Comment (muydanny) [inline spec/models/user_spec.rb:21]: Love this creativity!

File: app/controllers/cart_controller.rb [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
   def show
-    @items = cart.items
+    if current_user == nil || current_user.role != "admin"
+      @items = cart.items
+    else
+      redirect_to "/error404"
+    end
   end
 
   def empty
```
Comment (stellakunzang) [inline app/controllers/cart_controller.rb:21]: maybe not restful? *shrug*

**Review by muydanny — APPROVED** [review]

File: app/views/errors/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<h1>~ERROR 404 PAGE NOT FOUND~</h1>
+
+<h1> MAYBE YOU SHOULD JUST GALLUP ON OUT OF HERE. </h1>
```
Comment (sagemlee) [inline app/views/errors/show.html.erb:3]: Love this.

File: spec/features/navigation_spec.rb [hunk 4]
```diff
@@ -74,52 +74,200 @@
...
+      expect(current_path).to eq('/admin')
+
+      visit "/cart"
+
+      expect(current_path).to eq('/error404')
+
+      visit "/merchant"
+
+      expect(current_path).to eq('/error404')
+    end
```
Comment (stellakunzang) [inline spec/features/navigation_spec.rb:271]: SO MANY TESTS AMAZEBALLS

**Review by stellakunzang — APPROVED** [review]
So good!

**Pull request merged**